src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/ComponentViewLocationExpanderProvider.cs

```diff
@@ -56,6 +56,15 @@ public class ComponentViewLocationExpanderProvider : IViewLocationExpanderProvid
                         {
                             modulesWithComponentViews.Add(module);
                         }
+
+                        moduleComponentsViewFilePaths = fileProviderAccessor.FileProvider.GetViewFilePaths(
+                            module.SubPath + "/Pages/Shared/Components", RazorExtensions,
+                            viewsFolder: null, inViewsFolder: true, inDepth: true);
+
+                        if (moduleComponentsViewFilePaths.Any())
+                        {
+                            modulesWithComponentViews.Add(module);
+                        }
                     }
```

Comment:
Looks good but then you need to add the related locations as done for the SharedViewsPath ...

See my last comment in the thread